lib/adapters/class-searchpress.php

```diff
@@ -132,18 +132,28 @@ public function add_search_suggest_to_indexed_post_data( $data, $sp_post ) {
 
 	/**
 	 * A callback for the sp_search_query_args filter. Adds aggregations to the
-	 * request so that the response will include aggregation buckets.
+	 * request so that the response will include aggregation buckets, and
+	 * filters by any aggregations set.
 	 *
 	 * @param array $es_args The request args to be filtered.
 	 * @return array The filtered request args.
 	 */
 	public function add_aggs_to_es_query( $es_args ): array {
 		// Add our aggregations.
 		foreach ( $this->get_aggregations() as $aggregation ) {
+			// Add aggregations to the request so buckets will be returned with results.
 			$request = $aggregation->request();
 			if ( ! empty( $request ) ) {
 				$es_args['aggs'][ $aggregation->get_query_var() ] = $request;
 			}
+
+			// Add any set aggregations to the query so results are properly filtered.
+			if ( ! empty( $es_args['query']['bool']['must'] ) && is_array( $es_args['query']['bool']['must'] ) ) {
+				$filter = $aggregation->filter();
+				if ( ! empty( $filter ) ) {
+					$es_args['query']['bool']['must'] = array_merge( $es_args['query']['bool']['must'], $filter );
```

Comment:
`bool.filter` would be better than `bool.must`, as `bool.filter` is much cheaper. it should be OK if the original query uses `must`.